### ISSUE TITLE
Merge Hifimagnet toolboxes in Feel++ toolboxes

### ISSUE THREAD

**Issue by prudhomm** [feelpp/feelpp#856]
Currently the hifimagnet toolbox is maintained in hifimagnet. Already it uses the CSM toolbox. the thermoelectric toolbox should be used to. The magnetostatic toolbox could be merged once it is ready. 
This would be required to ease hifimagnet development in the long term, in hifimagnet would remain only the data sets (geometry json configurations...)

**Comment by prudhomm** [comment]
@romainhild @Trophime could you summarize the situation here ?
we are close to have this no ?

**Comment by romainhild** [comment]
We now have the three toolboxes needed, thermo-electric, maxwell, elasticity. But we lack some features that were implemented in hifimagnet, notably for the maxwell one.

There is also the question of the MagnetTools, they are used in hifimagnet to compute the magnetic-potential on surfaces to use as Dirichlet condition for the maxwell equations.
How can we manage them ?

So I think we can keep the issue open for now.

**Comment by prudhomm** [comment]
but magnet tools cannot go public, can they? @Trophime ? we _just_ have to provide the proper interface for them as inputs. 
Otherwise what is missing in maxwell exactly ?

**Comment by romainhild** [comment]
Yes indeed we need an interface with them.
For maxwell, we need to add:
- option for regularized/saddle-point formulation
- biotsavart boundary
- magnet tools boundary

Maybe other things @trophime?

**Comment by prudhomm** [comment]
Could you comment on what it means exactly?
if you have written down in your Phd this part I would be happy to read it

**Comment by romainhild** [comment]
Biotsavart/magnet tools boundary is what we talked about just before, use them to compute the magnetic potential on a surface to use as Dirichlet condition.
And for now, the maxwell toolbox only deals with the regularized formulation, we need to add the option to use the saddle point formulation.
And we also need to fix the bug with Nedelec in 3D and in parallel #1384

**Comment by prudhomm** [comment]
can we close this ? @romainhild @Trophime
